test/Microsoft.Health.Fhir.Shared.Tests.E2E/Rest/ConditionalDeleteTests.cs

```diff
@@ -0,0 +1,128 @@
+﻿// -------------------------------------------------------------------------------------------------
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License (MIT). See LICENSE in the repo root for license information.
+// -------------------------------------------------------------------------------------------------
+
+using System;
+using System.Linq;
+using System.Net;
+using System.Threading;
+using Hl7.Fhir.Model;
+using Microsoft.Health.Fhir.Client;
+using Microsoft.Health.Fhir.Core.Extensions;
+using Microsoft.Health.Fhir.Core.Features;
+using Microsoft.Health.Fhir.Core.Models;
+using Microsoft.Health.Fhir.Tests.Common;
+using Microsoft.Health.Fhir.Tests.Common.FixtureParameters;
+using Microsoft.Health.Fhir.Tests.E2E.Common;
+using Microsoft.Health.Fhir.Web;
+using Microsoft.Health.Test.Utilities;
+using Xunit;
+using Task = System.Threading.Tasks.Task;
+
+namespace Microsoft.Health.Fhir.Tests.E2E.Rest
+{
+    [HttpIntegrationFixtureArgumentSets(DataStore.All, Format.All)]
+    public class ConditionalDeleteTests : IClassFixture<HttpIntegrationTestFixture<Startup>>
+    {
+        private readonly TestFhirClient _client;
+
+        public ConditionalDeleteTests(HttpIntegrationTestFixture<Startup> fixture)
+        {
+            _client = fixture.TestFhirClient;
+        }
+
+        [Fact]
+        [Trait(Traits.Priority, Priority.One)]
+        public async Task GivenNoExistingResources_WhenDeletingConditionally_TheServerShouldReturnAccepted()
+        {
+            var identifier = Guid.NewGuid().ToString();
+            await ValidateResults(identifier, 0);
+
+            FhirResponse response = await _client.DeleteAsync($"{KnownResourceTypes.Observation}?identifier={identifier}", CancellationToken.None);
+            Assert.Equal(HttpStatusCode.NoContent, response.StatusCode);
+        }
+
+        [InlineData(true)]
+        [InlineData(false)]
+        [Theory]
+        [Trait(Traits.Priority, Priority.One)]
+        public async Task GivenOneMatchingResource_WhenDeletingConditionally_TheServerShouldDeleteSuccessfully(bool hardDelete)
+        {
+            var identifier = Guid.NewGuid().ToString();
+            await CreateWithIdentifier(identifier);
+            await ValidateResults(identifier, 1);
+
+            FhirResponse response = await _client.DeleteAsync($"{KnownResourceTypes.Observation}?identifier={identifier}&hardDelete={hardDelete}", CancellationToken.None);
```

Comment:
>hardDelete [](http://example.com/codeflow?start=121&length=10)

is it worth add multipleDelete here as well? For single case it should behave the same, why not check that?